crypto/txscript/examples/kip-10.rs

```diff
@@ -0,0 +1,675 @@
+use kaspa_addresses::{Address, Prefix, Version};
+use kaspa_consensus_core::{
+    hashing::{
+        sighash::{calc_schnorr_signature_hash, SigHashReusedValuesUnsync},
+        sighash_type::SIG_HASH_ALL,
+    },
+    tx::{
+        MutableTransaction, PopulatedTransaction, Transaction, TransactionId, TransactionInput, TransactionOutpoint,
+        TransactionOutput, UtxoEntry, VerifiableTransaction,
+    },
+};
+use kaspa_txscript::{
+    caches::Cache,
+    opcodes::codes::{
+        Op0, OpCheckSig, OpCheckSigVerify, OpDup, OpElse, OpEndIf, OpEqualVerify, OpFalse, OpGreaterThanOrEqual, OpIf, OpSub, OpTrue,
+        OpTxInputAmount, OpTxInputSpk, OpTxOutputAmount, OpTxOutputSpk,
+    },
+    pay_to_address_script, pay_to_script_hash_script,
+    script_builder::{ScriptBuilder, ScriptBuilderResult},
+    TxScriptEngine,
+};
+use kaspa_txscript_errors::TxScriptError::{EvalFalse, VerifyError};
+use rand::thread_rng;
+use secp256k1::Keypair;
+
+/// Main function to execute all Kaspa transaction script scenarios.
+///
+/// # Returns
+///
+/// * `ScriptBuilderResult<()>` - Result of script builder operations for all scenarios.
+fn main() -> ScriptBuilderResult<()> {
+    threshold_scenario()?;
+    threshold_scenario_limited_one_time()?;
+    threshold_scenario_limited_2_times()?;
+    shared_secret_scenario()?;
+    Ok(())
+}
+
+/// # Standard Threshold Scenario
+///
+/// This scenario demonstrates the use of custom opcodes and script execution within the Kaspa blockchain ecosystem.
+/// There are two main cases:
+///
+/// 1. **Owner case:** The script checks if the input is used by the owner and verifies the owner's signature.
+/// 2. **Borrower case:** The script allows the input to be consumed if the output with the same index has a value of input + threshold and goes to the P2SH of the script itself.
+///
+/// # Returns
+///
+/// * `ScriptBuilderResult<()>` - Result of script builder operations for this scenario.
+fn threshold_scenario() -> ScriptBuilderResult<()> {
+    println!("\n[STANDARD] Running standard threshold scenario");
+    // Create a new key pair for the owner
+    let owner = Keypair::new(secp256k1::SECP256K1, &mut thread_rng());
+
+    // Set a threshold value for comparison
+    let threshold: i64 = 100;
+
+    // Initialize a cache for signature verification
+    let sig_cache = Cache::new(10_000);
+
+    // Prepare to reuse values for signature hashing
+    let mut reused_values = SigHashReusedValuesUnsync::new();
```

Comment:
find and replace: `mut reused_values` -> `reused_values`
(essentially it seems like lints aren't running for this file )